### PR TITLE
Added poetry, and pre-commit hooks

### PR DESCRIPTION
**Pull request by nrbnlulu**
poetry -  should help you in the build process and pypi publishing.
pre-commit-hooks - catches many common developer mistakes and auto-formats. 

pre-commit catched many errors, If you would be interested in this PR i'll fix them and willing to collaborate on PyPi publishing. 

Best regards.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,185 @@
+[tool.poetry]
```
Comment (auscompgeek) [inline pyproject.toml:1]: If the package metadata is being moved into pyproject.toml, it should use PEP 621, not any legacy formats.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  | _build
+  | buck-out
+  | build
+  | dist
+)/
+'''
+
+[tool.ruff]
+line-length = 100
+select = ["ALL"]
```
Comment (auscompgeek) [inline pyproject.toml:55]: Do not enable `ALL`. We are not a Django project, nor are we a data science project.

**Review by auscompgeek — CHANGES_REQUESTED** [review]

File: cxxheaderparser/parser.py [hunk 3]
```diff
@@ -1,63 +1,60 @@
...
+from .types import TemplateParam
+from .types import TemplateSpecialization
+from .types import TemplateTypeParam
+from .types import Token
+from .types import Type
+from .types import Typedef
+from .types import UsingAlias
+from .types import UsingDecl
+from .types import Value
+from .types import Variable
```
[the remark this answers is not in the document]
Comment (nrbnlulu) [reply cxxheaderparser/parser.py:57]: https://github.com/asottile/reorder-python-imports#why-this-style

[the remark this answers is not in the document]
Comment (auscompgeek) [reply cxxheaderparser/parser.py:57]: I don't remember the last time we had merge conflicts in import statements in any RobotPy project.

**Comment by virtuald** [comment]
I don't really like poetry, and those "improved" imports are not improved.

**Comment by virtuald** [comment]
Also worth noting, the reason I haven't published on pypi yet is because I don't want to commit to the API yet until I've used this project on a real project, and I've been meaning to do so for.. 2 years now. Soon?

**Pull request closed without merging**